### PR TITLE
[Feat] Support Launch arguments from EXE

### PR DESCRIPTION
**Pull request by flavioislima**
This PR adds the ability for developers to pass Launch Arguments to their games directly from the EXE path.
So HP will understand if they do something like: `game.exe -arg1 -arg2` and will pass those arguments to the launch command.

QASE Test: https://app.qase.io/project/HC?case=334&suite=43&tab=&previewMode=modal

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Review by eliobricenov — APPROVED** [review]

**Review by red-game-dev — APPROVED** [review]

File: src/backend/storeManagers/storeManagerCommon/games.ts [hunk 1]
```diff
@@ -218,6 +218,17 @@ export function getGameProcessName(gameInfo: GameInfo): string | undefined {
...
+  executable: string
+  launchArgs: string
+} {
+  const match = executableWithArgs.match(/^(.*?\.(exe|app|bin|sh))/i)
+  const executable = match ? match[0] : ''
+  const launchArgs = executableWithArgs.replace(executable, '').trim()
+
+  return { executable, launchArgs }
+}
+
```
Comment (BrettCleary) [inline src/backend/storeManagers/storeManagerCommon/games.ts:231]: Can we move this to a separate module and add a unit test for it?

**Review by BrettCleary — APPROVED** [review]
code LGTM. I just really think we need some unit tests for `getExecutableAndArgs` since it's a pure function and a critical piece of code

**Comment by nyghtstalker** [comment]
I added the args to the executable path in the library.json and I am getting this error when I launch the game on Windows.

```
Error running command "LD_PRELOAD= "J:\Users\Joe\Games\HyperPlay\kosium\kosiumarcticeclipse\J:\Users\Joe\Games\HyperPlay\kosium\kosiumarcticeclipse\KosiumClient.exe -arg1 -arg2" -arg1 -arg2": Error: spawn J:\Users\Joe\Games\HyperPlay\kosium\kosiumarcticeclipse\KosiumClient.exe -arg1 -arg2 ENOENT
at ChildProcess._handle.onexit (node:internal/child_process:284:19)
at onErrorNT (node:internal/child_process:477:16)
at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

image
image

File: src/backend/storeManagers/storeManagerCommon/games.ts (hunk 1 above)
Comment (flavioislima) [reply src/backend/storeManagers/storeManagerCommon/games.ts]: done

**Comment by flavioislima** [comment]
@nyghtstalker the issue on windows should be fixed now 👍🏽

**Comment by nyghtstalker** [comment]
@flavioislima I am seeing the args being added to the path in the logs and terminal. Is this is expected and the goal of this PR? I want to make sure this is correct. If so, then great and I will approve it. Let me know!

image

**Comment by flavioislima** [comment]
> @flavioislima I am seeing the args being added to the path in the logs and terminal. Is this is expected and the goal of this PR? I want to make sure this is correct. If so, then great and I will approve it. Let me know!
> 
> image
> 

Yes, this and if the game is launching fine. 👍

**Review by nyghtstalker — APPROVED** [review]

**Pull request merged**